### ISSUE TITLE
Multiple partial pipeline: TypeError: fulfill_profile() got multiple values for keyword argument 'current_partial'

### ISSUE THREAD

**Issue by gnalbandian** [python-social-auth/social-core#286]
Hi,

I am having the following error when a second partial pipeline is reach in django.

I have a first partial pipeline method that goes fine, but when the second arrives I get:
TypeError: fulfill_profile() got multiple values for keyword argument 'current_partial'

I guess that every @partial adds a current_partial into kwargs.
Is this the way its intended?

Anyone got any idea how to solve this?

**Comment by omab** [comment]
There should be only one active partial at any time, once one is reached and continued, it's not called anymore, the partial works by storing the index in the pipeline, and when it's resumed, the pipeline starts from that position.

**Comment by gnalbandian** [comment]
I have two partial pipelines. The first one will ask to fill an email fields if the login backend response doesn't return one.

The second partial pipeline will ask for some additional data such as gender, nationality, etc, also if the backend response doesn't return those fields.

In some cases both are executed, sometimes none, or just one. It depends of the backend used to login.

The error happens when the second partial pipeline is reached. The two partial are not even consecutive. The first one.

The pipeline definition is: 
SOCIAL_AUTH_PIPELINE = (
    'social_core.pipeline.social_auth.social_details',
    'social_core.pipeline.social_auth.social_uid',
    'social_core.pipeline.social_auth.auth_allowed',
    'social_core.pipeline.social_auth.social_user',
    'socialLogin.pipeline.supplimentary_data.require_email', # Partial 1
    'social_core.pipeline.user.get_username',
    'social_core.pipeline.user.create_user',
    'social_core.pipeline.social_auth.associate_user',
    'social_core.pipeline.social_auth.load_extra_data',
    'socialLogin.pipeline.supplimentary_data.fulfill_profile',  # Partial 2
    'socialLogin.pipeline.user.user_details',

)

If i comment out one of them, that other works just fine.
The issues happens only when both are uncomment.

**Comment by gnalbandian** [comment]
Hi @omab. Didyou understood what I meant? Were you able to recreate the issue?

**Comment by omab** [comment]
@gnalbandian I did understand the case and confuses me a bit because multiple partial steps are a very common scenario which I have implemented in the past as well, still will take a look.

**Comment by gnalbandian** [comment]
Thank you @omab .
 I'm from Argentina, maybe if you prefer we can get in touch through skype or hangouts so as to show you exactly the issue.

Anyway, This is my requiere_email, just the same as yours:
`@partial
def require_email(strategy, details, user=None, is_new=False, *args, **kwargs):
    if kwargs.get('ajax') or user and user.email:
        return
    elif is_new and not details.get('email'):
        email = strategy.request_data().get('email')
        if email:
            details['email'] = email
        else:
            current_partial = kwargs.get('current_partial')
            return strategy.redirect(
                '/email?partial_token={0}'.format(current_partial.token)
            )`

And my fullfil_profile:
`def fulfill_profile(strategy, details, backend, user=None, is_new=False, *args, **kwargs):
    if kwargs.get('ajax') or user and user.email:
        return kwargs
    

    supp_data_required = False
    

	birth_date = strategy.request_data().get('birth_date')
	
	if birth_date:
		details['birth_date'] = birth_date
	else:
		supp_data_required = True
	
	if supp_data_required:
            current_partial = kwargs.get('current_partial')
            return strategy.redirect('/suppdata?partial_token={0}'.format(current_partial.token))

    return kwargs`

**Comment by omab** [comment]
@gnalbandian, I've tested the scenario using [social-examples](https://github.com/python-social-auth/social-examples), I've just pushed a change adding more partial-pipeline steps to the apps in there. Still, no issues were raised, the flow works as expected.

I've noted that your `fulfill_profile` returns `kwargs`, that dict contains the `current_partial` which is then cached in the partial pipeline table (or session), so when the partial is continued, it will be pulled and added to the parameters getting passed, and clashes with the `current_partial` instance. I'm not sure what's your intention for returning `kwargs` there, but that's the reason for the error.
